user_manual/ada_api_tutorial.rst

```diff
@@ -396,3 +401,92 @@ name resolution in Ada:
 
 You can find these and all the other properties documented in your favorite
 language's API reference.
+
+.. _ada-generic-app:
+
+Ada generic application framework
+=================================
+
+In order to facilitate the creation of Ada command line applications,
+we have added an ``App`` generic package, that you can simply
+instantiate in order to create a command line application with a lot
+of common functionality already built-in, so that you don't have to reinvent it
+every time.
+
+The way it works is that it handles all the logic for you. You just pass it one
+or two callbacks:
+
+- One callback, `Process_Unit`, will be called for each new `AnalysisUnit` that
```

Comment:
This tutorial is Ada-specific, so I suggest to spell `Analysis_Unit` (same below). Also, this is Sphinx, so code should be surrounded with double backquotes (here and several times below).